tests/ui/mir/issue-109567.rs

```diff
@@ -0,0 +1,14 @@
+// run-pass
+// check-run-results
+
+pub fn f() -> f64 {
+    std::hint::black_box(-1.0) % std::hint::black_box(-1.0)
```

Comment:
What part of the compiler are you specifically trying to test here?

If you're updating const eval, maybe it would make sense to add a test for this in <https://github.com/rust-lang/rust/tree/master/tests/mir-opt/const_prop>?

Reply:
The new fix don't change `mir` part, so I moved the test case to `tests/ui/numbers-arithmetic/`.